### DISCUSSION TITLE
shortest_simple_paths with negative weights?

### DISCUSSION THREAD

**Discussion by aacifuentesb** [networkx/networkx#6826] [Q&A]
Hello!

Searching for an algorithm capable of finding the k-shortest simple path for a DiGraph that has some edges with negative weights and has no cycles, i found the shortest_simple_paths() function. In the documentation says:

> If a weighted shortest path search is to be used, no negative weights are allowed.

Is there any algorithm implemented for this problem, maybe using bellman-ford as a shortest path finder for my case?  There's an issue #3685 where they mention this case, but no solution is implemented. Maybe using the Potential idea of Johnson's algorithm to make the weights positive?

Greetings!

**Comment by Transurgeon** [comment]
I remember finding a paper for the shortest simple path with negative weights [here](https://arxiv.org/abs/2203.03456). 
There is also a quanta magazine article that gives an overview of the algorithm [here](https://www.quantamagazine.org/finally-a-fast-algorithm-for-shortest-paths-on-negative-graphs-20230118/).
Finally, there is also some code implementation (in java :( ) that you can find [here](https://github.com/nevingeorge/Negative-Weight-SSSP). 
I have never used this code or even read this paper, so cannot help further than that. It seems like the algorithm requires many relaxations and I am not sure if you will be able to implement it for your use case. 
Anyhow, to my current knowledge, networkx doesn't have support for this yet.
